partiql-eval/src/eval.rs

```diff
@@ -101,6 +103,9 @@ pub enum EvaluationError {
 pub trait Evaluable: Debug {
     fn evaluate(&mut self, ctx: &dyn EvalContext) -> Option<Value>;
     fn update_input(&mut self, input: &Value, branch_num: u8);
+    fn get_vars(&self) -> Vec<String> {
```

Comment:
This is used for `LEFT JOIN`s when the evaluation of the RHS returns an empty bag of bindings. From [section 5.4 of the spec](https://partiql.org/assets/PartiQL-Specification.pdf#subsection.5.4), when this case occurs, we create binding with every variable defined from the RHS `BindingOp` (e.g. the `AS` and `AT` bindings for `Scan` and `Unpivot`). Not sure if there was a better way to get the variables defined by a `BindingOp`.